### ISSUE TITLE
[EDC 0.7.1] Shell Lookup from EDC Data Plane

### ISSUE THREAD

**Issue by tanweersalah** [eclipse-tractusx/tractusx-edc#1345]
EDC  == 0.7.1

While requesting the Catalog from registry for a particular assetId, I am getting the empty list, 

while requesting directly to DTR I am getting the response, 

<img width="1145" alt="Screenshot 2024-06-01 at 12 31 29 PM" src="https://github.com/eclipse-tractusx/digital-product-pass/assets/31222891/178666aa-f3b1-4207-bab6-4ec9f1b6631f">

But when the same assetid is being lookup from the EDC Data Plane , I am getting empty result. 

<img width="1165" alt="Screenshot 2024-06-01 at 12 31 54 PM" src="https://github.com/eclipse-tractusx/digital-product-pass/assets/31222891/56c1339f-121b-461e-9e1c-a5548a5d6e7b">

**Comment by tanweersalah** [comment]
here is the congfiguration for the registry-assest : 

`
{
    "description": [],
    "displayName": [],
    "id": "urn:uuid:6bb0929b-8f29-cc77-0975-ec30a0e2415e",
    "specificAssetIds": [
        {
            "supplementalSemanticIds": [],
            "name": "manufacturerPartId",
            "value": "XYZ78901",
            "externalSubjectId": {
                "type": "ExternalReference",
                "keys": [
                    {
                        "type": "GlobalReference",
                        "value": "PUBLIC_READABLE"
                    }
                ]
            }
        }
    ],
    "submodelDescriptors": [
        {
            "endpoints": [
                {
                    "interface": "SUBMODEL-3.0",
                    "protocolInformation": {
                        "href": "http://supplier-control-plane:9184/api/public/data/urn:uuid:79f4649a-ce85-bda6-fcb4-a172df0beba8",
                        "endpointProtocol": "HTTP",
                        "endpointProtocolVersion": [
                            "1.1"
                        ],
                        "subprotocol": "DSP",
                        "subprotocolBody": "id=urn:uuid:0c3d2db0-e5c6-27f9-5875-15a9a00e7a27;dspEndpoint=http://supplier-control-plane:9184",
                        "subprotocolBodyEncoding": "plain",
                        "securityAttributes": [
                            {
                                "type": "NONE",
                                "key": "NONE",
                                "value": "NONE"
                            }
                        ]
                    }
                }
            ],
            "idShort": "batteryPass",
            "id": "urn:uuid:79f4649a-ce85-bda6-fcb4-a172df0beba8",
            "semanticId": {
                "type": "ExternalReference",
                "keys": [
                    {
                        "type": "Submodel",
                        "value": "urn:bamm:io.catenax.battery.battery_pass:3.0.1#BatteryPass"
                    }
                ]
            },
            "supplementalSemanticId": [],
            "description": [
                {
                    "language": "en",
                    "text": "Battery Passport Submodel"
                }
            ],
            "displayName": []
        }
    ]
}
`

**Comment by wolf4ood** [comment]
Hi @tanweersalah 

this is for the DTR registry right? do you have the asset payload on EDC side?

Thanks

**Comment by tanweersalah** [comment]
@wolf4ood , sorry I did not understand what you meant by having asset payload on edc side.

So let me rephrase the issue, 

After EDR negotiation the first EDR receive is for the DTR Asset, 

I am getting empty response when the assetid is being lookup from Dataplane of provider, while the same asset id I am able to fetch when using the DTR endpoints.

**Comment by tanweersalah** [comment]
Also i am able to fetch all shell descriptors from Dataplane , just when I look for specific Asset Id , I am getting empty result. 

<img width="1075" alt="Screenshot 2024-06-03 at 11 20 35 AM" src="https://github.com/eclipse-tractusx/tractusx-edc/assets/31222891/300cd083-89e6-465b-8f47-e7bf3086271c">

**Comment by wolf4ood** [comment]
@tanweersalah 

is the provider in your control? did you configure the asset on the EDC side?

**Comment by tanweersalah** [comment]
@wolf4ood yes I have the access of provider edc. And configured the asset on the provider side.

**Comment by wolf4ood** [comment]
Can you share the asset payload that you used for configuring the DTR registry asset on provider side?

Thanks

**Comment by tanweersalah** [comment]
```
{
    "@id": "registry-asset",
    "@type": "Asset",
    "properties": {
        "version": "3.0",
        "type": {
            "@id": "DigitalTwinRegistry"
        },
        "asset:prop:type": "data.core.digitalTwinRegistry",
        "id": "registry-asset"
    },
    "dataAddress": {
        "@type": "DataAddress",
        "proxyPath": "true",
        "type": "HttpData",
        "proxyMethod": "true",
        "proxyQueryParams": "true",
        "proxyBody": "true",
        "baseUrl": "http://dtr-supplier:4243/api/v3"
    },
    "@context": {
        "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
        "edc": "https://w3id.org/edc/v0.0.1/ns/",
        "tx": "https://w3id.org/tractusx/v0.0.1/ns/",
        "tx-auth": "https://w3id.org/tractusx/auth/",
        "cx-policy": "https://w3id.org/catenax/policy/",
        "odrl": "http://www.w3.org/ns/odrl/2/"
    }
}
```

**Comment by wolf4ood** [comment]
Can you also check on EDC provider side logs when you are fetching the data it should log the entire URL of the translated URL of the DTR registry.

**Comment by tanweersalah** [comment]
```
supplier-data-plane     | DEBUG 2024-06-03T09:47:53.911021264 Executing HTTP request: http://dtr-supplier:4243/api/v3/lookup/shells?assetIds=eyJuYW1lIjoicGFydEluc3RhbmNlSWQiLCJ2YWx1ZSI6IklNUjE4NjUwVjEifQ==
```

**Comment by wolf4ood** [comment]
Cool so from the EDC point of view all seems correct.

Is there some kind of filtering on DTR registry based on headers? 

I remember that the registry filtered the requests based on `Edc-Bpn` header that the EDC will forward on the HTTP request

**Comment by tanweersalah** [comment]
Yes For DTR request ,

Edc-Bpn :  {{bpnNumberProvider}}

need to be added in header, 

Does the edc adds it internally ??

**Comment by wolf4ood** [comment]
Yes it attaches to the request

**Comment by tanweersalah** [comment]
is there any way to check what bpn it is attaching ?

**Comment by wolf4ood** [comment]
It attaches the consumer BPN

**Comment by tanweersalah** [comment]
Ok I got the issue, i created the shel in registry with providerbpn , 

Now after updating it to consumer bpn I am getting the results. 

So doest it mean for every consumer we have to create a shell descriptor in DTR ?

**Comment by wolf4ood** [comment]
I'm no expert on DTR and how the visibility restriction and configuration works there

**Comment by tanweersalah** [comment]
@wolf4ood  No problem and Thanks for the prompt responses. Have a nice day.

**Comment by wolf4ood** [comment]
:+1: 

thanks @tanweersalah , closing this then :)
